app/assets/javascripts/discourse/app/components/post/menu/buttons/like.gjs

```diff
@@ -82,17 +92,112 @@ export default class PostMenuLikeButton extends Component {
       </div>
     {{else}}
       <div class="double-button">
-        <LikeCount
-          ...attributes
-          @action={{@buttonActions.toggleWhoLiked}}
-          @state={{@state}}
-          @post={{@post}}
-        />
+        {{#if @post.likeCount}}
+          <LikedUsersList ...attributes @post={{@post}} />
+        {{else}}
+          <LikeCount
+            ...attributes
+            @action={{@buttonActions.toggleWhoLiked}}
+            @state={{@state}}
+            @post={{@post}}
+          />
+        {{/if}}
       </div>
     {{/if}}
   </template>
 }
 
+class LikedUsersList extends Component {
+  @service store;
+
+  @tracked likedUsers = null;
+  @tracked loadingLikedUsers = false;
+
+  @action
+  async fetchLikedUsers() {
+    if (this.likedUsers || this.loadingLikedUsers) {
+      return;
+    }
+
+    this.loadingLikedUsers = true;
+
+    try {
+      const users = await this.store
+        .find("post-action-user", {
+          id: this.args.post.id,
+          post_action_type_id: 2, // LIKE_ACTION
+        })
+        .then((result) => result.toArray());
+
+      this.likedUsers = users;
+    } catch {
+    } finally {
+      this.loadingLikedUsers = false;
+    }
+  }
+
+  <template>
+    <DMenu
+      @modalForMobile={{true}}
+      @identifier="post-like-users"
+      @triggers="click"
+      @onShow={{this.fetchLikedUsers}}
+      @triggerClass="button-count"
+      @placement="top"
+    >
+      <:trigger>
+        {{@post.likeCount}}
+      </:trigger>
+      <:content>
+        <ConditionalLoadingSpinner
+          @condition={{this.loadingLikedUsers}}
+          class="liked-users-list__container"
+        >
+          <span class="liked-users-list__count">
+            {{icon "d-liked" class="liked-users-list__count-icon"}}
+            {{@post.likeCount}}
+          </span>
+          <ul class="liked-users-list">
+            {{#each this.likedUsers as |user|}}
+              <li class="liked-users-list__item">
+                <LikedUserItem @user={{user}} />
+              </li>
+            {{/each}}
+          </ul>
+        </ConditionalLoadingSpinner>
+      </:content>
+    </DMenu>
+  </template>
+}
+
+class LikedUserItem extends Component {
```

Comment:
Could `UserAvatar` component be used instead of this implementation?

Reply:
Yes it could, thanks! I was trying to use `{{UserAvatar...` and kept encountering errors. `<UserAvatar />` works perfect.